src/core/client/stream/mutations/CreateCommentReplyMutation.ts

```diff
@@ -10,35 +11,44 @@ import { TalkContext } from "talk-framework/lib/bootstrap";
 import {
   commitMutationPromiseNormalized,
   createMutationContainer,
+  MutationInput,
+  MutationResponsePromise,
 } from "talk-framework/lib/relay";
-import { Omit } from "talk-framework/types";
 import { CreateCommentReplyMutation as MutationTypes } from "talk-stream/__generated__/CreateCommentReplyMutation.graphql";
 
 import {
   incrementStoryCommentCounts,
+  isInReview,
+  isRejected,
+  isRolePriviledged,
   prependCommentEdgeToProfile,
 } from "../helpers";
 
-export type CreateCommentReplyInput = Omit<
-  MutationTypes["variables"]["input"],
-  "clientMutationId"
-> & { local?: boolean };
+export type CreateCommentReplyInput = MutationInput<MutationTypes> & {
+  local?: boolean;
+};
 
 function sharedUpdater(
   environment: Environment,
   store: RecordSourceSelectorProxy,
   input: CreateCommentReplyInput
 ) {
+  const commentEdge = store
+    .getRootField("createCommentReply")!
+    .getLinkedRecord("edge")!;
+  const status = commentEdge.getLinkedRecord("node")!.getValue("status");
+
+  // If comment is in review or has been rejected, we don't need to add it.
+  if (isInReview(status) || isRejected(status)) {
```

Comment:
Would it not be better here to rephrase this to say, `isVisable` instead? This way we can have more granular control over when a comment is visible instead of these special cases.

Reply:
Makes sense 😄